### DISCUSSION TITLE
Native portable mode: store settings in winmerge.ini file instead of the registry

### DISCUSSION THREAD

**Discussion by eadmaster** [WinMerge/winmerge#1601] [Ideas]
I know there's the [PortableApps package](https://portableapps.com/apps/utilities/winmerge_portable), but i'd like to have native portable mode without using a launcher.

**Comment by sdottaka** [comment]
If you create an empty winmerge.ini file in the folder where WinMergeU.exe exists, WinMerge will read and write settings from that winmerge.ini file.

However, there are exceptions, and please note that settings for each plugin and shell extension information are not written to winmerge.ini but written to the registry.

**Reply by kazumori102** [reply]
Regarding the statement that "settings for each plugin are written to the registry":
One of the reasons why portable applications are desired/sought is the ease of machine migration/transfer.
Could you please specify which registry entries need to be preserved?
Alternatively, would it be possible to implement an alternative method for storing these settings that doesn't depend on the registry?

*Note: I fully understand that shell extensions structurally require registry usage.

**Reply by sdottaka** [reply]
Plugin settings are normally stored in the registry under:
 `HKEY_CURRENT_USER\Software\Thingamahoochie\WinMerge\Plugins`

However, newer plugins save their settings in the INI file used by WinMerge — either `winmerge.ini` located next to `WinMergeU.exe`, or the INI file specified with the `/INI` command line option.

Shell extensions still require registry entries.

**Reply by kazumori102** [reply]
I've confirmed that the file save location for plugin user settings is limited to either `AppData` or the `Documents` folder.
This is not ideal. Please make it possible to select an arbitrary location, or alternatively `${WINMERGE_HOME}\MergePlugins\`.
I believe I might be able to modify this myself if I have time, but I would appreciate it if you could consider this enhancement.
